### PR TITLE
vlan translation

### PR DESCRIPTION
**Pull request by grzegorzkisiel**

### CODE REVIEW

File: ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/activator/OvsActivatorHelper.java [hunk 1]
```diff
@@ -163,4 +168,18 @@ private IngressBwpFlow getIngressBwpFlow() {
     protected boolean isIBwpConfigured() {
         return getIngressBwpFlow() != null;
     }
+
+    public static void validateExternalVLANs(List<EndPoint> endPoints) throws ResourceNotAvailableException {
+        final Set<Integer> vlans = new HashSet<>();
+        endPoints.stream().forEach(endPoint -> {
```
Comment (bartoszm) [inline ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/activator/OvsActivatorHelper.java]: if you use streams to do that, please use 
map method
ePs.stream.map{eP -> id).filter(id >=0).collect(toSet)

File: impl/src/main/java/org/opendaylight/unimgr/mef/nrp/common/NrpDao.java [hunk 2]
```diff
@@ -79,21 +82,31 @@ public Node createNode(String topologyId, String nodeId, LayerProtocolName name,
...
-                .build();
-        tx.put(LogicalDatastoreType.OPERATIONAL, node(new Uuid(topologyId), new Uuid(nodeId)), node);
+                .addAugmentation(Node1.class, new Node1Builder().setActivationDriverId(activationDriverId).build());
+
+        Node node = serviceVlanMapList == null ? nb.build() : nb.addAugmentation(Node2.class, new Node2Builder().setServiceVlanMap(serviceVlanMapList).build()).build();
+        tx.put(LogicalDatastoreType.OPERATIONAL, node(nodeId), node);
         return node;
     }
 
+    public void updateNode(Node node) {
```
Comment (bartoszm) [inline impl/src/main/java/org/opendaylight/unimgr/mef/nrp/common/NrpDao.java]: this is in fact an override of a node, how about renaming operation?
please bare in mind that all external changes between reading/modyfying the node given as parameter and writing it are silently lost

File: impl/src/main/java/org/opendaylight/unimgr/mef/nrp/impl/topologytervice/TapiTopologyServiceImpl.java [hunk 3]
```diff
@@ -13,6 +13,8 @@
 import org.opendaylight.controller.md.sal.common.api.data.LogicalDatastoreType;
 import org.opendaylight.controller.md.sal.common.api.data.ReadFailedException;
 import org.opendaylight.unimgr.mef.nrp.common.NrpDao;
+import org.opendaylight.yang.gen.v1.urn.odl.unimgr.yang.unimgr.ext.rev170531.Node1;
+import org.opendaylight.yang.gen.v1.urn.odl.unimgr.yang.unimgr.ext.rev170531.Node2;
```
Comment (bartoszm) [inline impl/src/main/java/org/opendaylight/unimgr/mef/nrp/impl/topologytervice/TapiTopologyServiceImpl.java]: maybe we could start naming the augmentations unimgr-ext is our file so we can do that 

to do that you need to import yang-ext and use ext:augment-identifier "xxx" then instead Node1... you will have XXX

File: ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/util/OpenFlowUtils.java [hunk 4]
```diff
@@ -195,34 +196,35 @@ private static Flow createDefaultFlow() {
...
+    private static Flow createVlanPassingFlow(String outputPort, String inputPort, int internalVlanId, int externalVlanId, String serviceName) {
         // Create list of actions and VLAN match
         List<Action> actions = new ArrayList<>();
-        Match vlanMatch;
         int actionOrder = 0;
 
-        vlanMatch = MatchUtils.createVlanMatch(internalVlanId, inputPort);
-
         actions.add(ActionUtils.createPopVlanAction(actionOrder++));
+        if(externalVlanId>=0){
```
Comment (bartoszm) [inline ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/util/OpenFlowUtils.java]: why do we need to check that? (as well in other places)

File: ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/util/VlanUtils.java [hunk 5]
```diff
@@ -35,93 +40,64 @@
...
     private final static Logger LOG = LoggerFactory.getLogger(VlanUtils.class);
 
-    public VlanUtils(List<NullAwareDatastoreGetter<Node>> nodes) throws ResourceNotAvailableException {
-        getAllVlanIDs(nodes);
+
+    public VlanUtils(DataBroker dataBroker, String nodeId ) throws ResourceNotAvailableException {
+        this.dataBroker = dataBroker;
+        try {
+            node = new NrpDao(dataBroker.newReadOnlyTransaction()).getNode(new Uuid(nodeId));
+            if (node == null) throw new ReadFailedException(null);
```
Comment (bartoszm) [inline ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/util/VlanUtils.java]: why not to throw ResoureceNotAvailable directly?

**Review by bartoszm — CHANGES_REQUESTED** [review]

File: ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/activator/OvsActivatorHelper.java (hunk 1 above)
Comment (grzegorzkisiel) [reply ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/activator/OvsActivatorHelper.java]: will be rewritten

File: impl/src/main/java/org/opendaylight/unimgr/mef/nrp/common/NrpDao.java (hunk 2 above)
Comment (grzegorzkisiel) [reply impl/src/main/java/org/opendaylight/unimgr/mef/nrp/common/NrpDao.java]: it was modeled on method updateNep.
the name can be changed. Is 'putNode' ok ?

File: ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/util/OpenFlowUtils.java (hunk 4 above)
Comment (grzegorzkisiel) [reply ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/util/OpenFlowUtils.java]: When externalVlanId is -1 it means that it is port-base service and then no vlan is poped on input port and added on outport

File: ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/util/VlanUtils.java (hunk 5 above)
Comment (grzegorzkisiel) [reply ovs-driver/src/main/java/org/opendaylight/unimgr/mef/nrp/ovs/util/VlanUtils.java]: will be corrected

**Pull request merged**